polo-bukkit/src/main/java/dev/dhdf/polo/bukkit/Main.java

```diff
@@ -94,6 +101,66 @@ public void broadcastMessage(String message) {
         this.getServer().broadcastMessage(message);
     }
 
+    @Override
+    public void kickPlayer(UUID uuid, String reason, String source) {
+        Player player = this.getServer().getPlayer(uuid);
+        if (player == null) {
+            Logger logger = getLogger();
+            logger.warning("No player to kick with UUID " + uuid.toString());
+            return;
+        }
+        getServer().getScheduler().runTask(this, new Runnable() {
+            public void run() {
+                player.kickPlayer(reason);
+                getServer().broadcast(source + " kicked " + player.getName() + " for " + reason, "matrix.kick.notify");
+            }
+        });
+    }
+
+    @Override
+    public void banPlayer(UUID uuid, String reason, String source) {
+        Logger logger = getLogger();
+        OfflinePlayer offlinePlayer = getServer().getOfflinePlayer(uuid);
+        if (offlinePlayer == null) {
+            logger.warning("No player to ban with UUID " + uuid.toString());
+            return;
+        }
+        BanList banlist = getServer().getBanList(BanList.Type.NAME);
+        if (banlist == null) {
+            logger.warning("No banlist found");
+            return;
+        }
+        banlist.addBan(offlinePlayer.getName(), reason, null, source);
+        getServer().broadcast(source + " banned " + offlinePlayer.getName() + " for " + reason, "matrix.ban.notify");
+
+        Player player = getServer().getPlayer(uuid);
```

Comment:
Get the player from `offlinePlayer` to avoid querying the server twice.
```suggestion
        Player player = offlinePlayer.getPlayer();
```